proto/buf/alpha/registry/v1alpha1/repository_branch.proto

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.alpha.registry.v1alpha1;
+
+import "google/protobuf/timestamp.proto";
+
+message RepositoryBranch {
+  // The name of the repository branch.
+  string name = 1;
+  // The name of the author.
+  string author_name = 2;
```

Comment:
Right, I should make it more clear. How does `latest_git_commit_author_name` sound?

Reply:
That seems reasonable to me.

Reply:
@bufdev are you okay with this?

Reply:
nit: Maybe `RepositoryCommit latest`? I can see this more generally useful, and `RepositoryCommit` has `GitCommitMetadata git_metadata`.

Reply:
Hmm -- so I think the metadata we're sending through is fine, in that yes it would have extra things like tag, etc., but you are still getting the correct git metadata because it is associated with the git commit mapped to the head of this branch. I think it's more, do we see this endpoint needing more git metadata down the road/expanding the UI around this? TBH, I don't think I'm that concerned about the data size, but it is adding a lot more information than we initially need.

Reply:
Related is that the `update_time` field is also implicitly coming from the `RepositoryCommit`/`GitMetadata` depending on whether we're showing commit time or push time.
But thinking about this differently, what we've got seems to be frontend-driven, versus we also want the CLI to be able to list branches at some point, and it's convenient to have the data for the head of a branch available immediately.
If this is not intended to be general-purpose/is frontend-driven, I'm cool with what you have now :)

Reply:
I was only considering the frontend, but if we are certain of its use case in the CLI, it'd certainly be nice to made general-purpose from the beginning. On further investigation, it appears to get quite complicated to include a full commit type in branches as it involves making multiple table joins and queries to include nested `RepositoryTag` and `GitCommitMetadata`, which also includes nested tags.  
1. If we strongly feel the need to make it general-purpose to serve CLI, invest in the time to include the commit. (I'm not sure about this one, so I'll lean on you @saquibmian and @doriable.)
2. Include minimum field necessary to serve the known needs for the frontend at this point with a room for expandability. We could go as simple as adding only the `head_commit_name` field for now, which can be used by the frontend to get additional metadata for the commit if needed.

Reply:
Based on chat with @doriable and @saquibmian over slack, going with #2 to fulfill the current frontend requirement, the latest_commit_name to be able to get commit metadata as requested, and append other metadata as needed in the future.

Reply:
I'm onboard with this approach, going to approve the change :)